envpool/sokoban/level_loader.h

```diff
@@ -41,16 +42,17 @@ class LevelLoader {
   int levels_loaded_{0};
   int env_id_{0};
   int num_envs_{1};
-  std::vector<SokobanLevel> levels_{0};
-  int cur_level_;
+  std::vector<std::pair<int, SokobanLevel>> levels_{0};
+  int cur_level_{-1}, cur_level_file_{-1};
   std::vector<std::filesystem::path> level_file_paths_{0};
   std::vector<std::filesystem::path>::iterator cur_file_;
   void LoadFile(std::mt19937& gen);
 
  public:
   int verbose;
 
-  std::vector<SokobanLevel>::iterator GetLevel(std::mt19937& gen);
+  std::pair<std::vector<std::pair<int, SokobanLevel>>::iterator, int> GetLevel(
```

Comment:
I know this is a relatively simple data structure ( `tuple[list[tuple[int, Level]], int]` ), but I think this would benefit a lot from having names. Probably what I'd do is:

```cpp
struct TaggedSokobanLevel {
  int file_idx, level_idx;
  SokobanLevel data;
}
```

which would make it much easier to parse what's the output of this and what's going on in `level_loader.cc`.


Also minimizes refactoring. It would also be a possibility to name this `struct SokobanLevel` and make the original one `SokobanLevelData`. Idk.